### PR TITLE
{cfd} [intel/2016b] OpenFOAM/4.0 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
Needs ~~#3440~~, ~~#3407~~ and ~~#3460~~ and
~~hpcugent/easybuild-easyblocks#986~~

### CODE REVIEW

**Comment by wpoely86** [comment]
@boegel ready for review

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2613.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/75203676fa07f9c3c7036a40c1556b99 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 6 out of 9 (8 easyconfigs in this PR)
node2074.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/13e820698aa4521220665722fb4069af for a full test report.

File: easybuild/easyconfigs/m/METIS/METIS-5.1.0-use-doubles.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Use double for floating point (64 bit) in METIS by default
```
Comment (boegel) [inline easybuild/easyconfigs/m/METIS/METIS-5.1.0-use-doubles.patch]: @wpoely86 add author please, and maybe also mention *why* you're making this chance (required by OpenFOAM 4.0?)

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+download_suffix = 'download.php?submit=Download&version=v%(version_major_minor)s&type=source&os=all&downloadFile='
+source_urls = ['http://www.paraview.org/paraview-downloads/%s' % download_suffix]
+sources = ["ParaView-v%(version)s.tar.gz"]
+
+dependencies = [
+    ('X11', '20160819'),
+    ('Mesa', '12.0.1'),
+    ('libGLU', '9.0.0'),
+    ('Qt', '4.8.7'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: @wpoely86 `Qt` 4.x here, but `Qt5` as dep for OpenFOAM, is that intentional?

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+]
+
+builddependencies = [('CMake', '3.6.1')]
+
+separate_build_dir = True
+
+configopts = '-DPARAVIEW_INSTALL_DEVELOPMENT_FILES=ON -DVTK_OPENGL_HAS_OSMESA=ON -DPARAVIEW_USE_MPI=ON '
+configopts += '-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include -DOPENGL_gl_LIBRARY=$EBROOTMESA/lib/libGL.so '
+configopts += '-DOSMESA_INCLUDE_DIR=$EBROOTMESA/include -DOSMESA_LIBRARY=$EBROOTMESA/lib/libOSMesa.so '
+configopts += '-DOPENGL_glu_LIBRARY=$EBROOTLIBGLU/lib/libGLU.so '
```
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: @wpoely86 please use `SHLIB_EXT`

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+configopts += '-DVTK_USE_SYSTEM_HDF5=ON -DPARAVIEW_ENABLE_PYTHON=ON -DBUILD_SHARED_LIBS=ON '
+# Without internet connection turn off testing (uncomment the following line)
+configopts += '-DBUILD_TESTING=OFF '
+#configopts += '-DVTK_USE_X=OFF '
```
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: @wpoely86 just remove this? or specify `ON` explicitly?

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: yes, building ParaView with Qt5 should be possible but all official releases are still with 4.x
Last time I tried 5.x it gave trouble.

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 4 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: I should add a note that this is for if you want to build ParaView server only

Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: OK, but do also set this to `ON` explicitely

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: OK, maybe add a comment on that?

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 4 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: why? It's on all right by default.

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: it's all on the paraview wiki.

File: easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/p/ParaView/ParaView-5.1.2-intel-2016b-mpi.eb]: being explicit doesn't hurt?

**Comment by boegel** [comment]
lgtm, should be good to go if test reports work out

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2612.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/db36424f2bed4b3be808c4c874e55371 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2025.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/49bb02ea220c5e2f652cced59f17fc1a for a full test report.

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**